packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php

```diff
@@ -63,16 +64,22 @@ public function shouldSkipReturnTypeChange(ClassMethod $classMethod, Type $paren
         return $this->typeComparator->areTypesEqual($currentReturnType, $parentType);
     }
 
-    private function resolveParentClassMethod(ClassMethod $classMethod): ?MethodReflection
+    private function resolveParentClassMethod(ClassMethod|MethodReflection $classMethod): ?MethodReflection
     {
-        $classReflection = $this->reflectionResolver->resolveClassReflection($classMethod);
-        if (! $classReflection instanceof ClassReflection) {
-            // we can't resolve the class, so we don't know.
-            throw new UnresolvableClassException();
+        if ($classMethod instanceof ClassMethod) {
+            $classReflection = $this->reflectionResolver->resolveClassReflection($classMethod);
+            if (! $classReflection instanceof ClassReflection) {
+                // we can't resolve the class, so we don't know.
+                throw new UnresolvableClassException();
```

Comment:
can this return null? throw exception may cause random error on overlap Scope filling, eg: reprint usage

Reply:
this exception is catched in the 2 call-site of this method. we added it pretty recently in

Reply:
sorry - I don't get what you are trying to tell me. could you elaborate?

Reply:
Oh, I see, it already covered in the caller, looking good then 👍